holidays/countries/italy.py

```diff
@@ -263,419 +289,666 @@ def __init__(self, *args, **kwargs):
+        # Restored by Royal Decree-Law 2859 of Dec 30, 1923.
+        # Abolished by Law 54 of Mar 5, 1977.
+        if self._year <= 1913 or 1924 <= self._year <= 1976:
             # Saints Peter and Paul.
-            self._add_saints_peter_and_paul_day("Santi Pietro e Paolo")
+            self._add_saints_peter_and_paul_day(tr("Santi Apostoli Pietro e Paolo"))
 
             # Corpus Christi.
-            self._add_corpus_christi_day("Corpus Domini")
+            self._add_corpus_christi_day(tr("Corpus Domini"))
+
+        # Established by Royal Decree 5342 of Oct 17, 1869.
 
         # Assumption Of Mary Day.
-        self._add_assumption_of_mary_day("Assunzione della Vergine")
+        self._add_assumption_of_mary_day(tr("Assunzione della Beata Vergine Maria"))
+
+        # Established by Royal Decree 5342 of Oct 17, 1869.
+        # Abolished by Royal Decree 1027 of Aug 4, 1913.
+        if self._year <= 1912:
+            # Nativity of Mary.
+            self._add_nativity_of_mary_day(tr("Natività della Beata Vergine Maria"))
+
+        # Established by Law 401 of Jul 19, 1895.
+        # Abolished by Law 1726 of Dec 27, 1930.
+        if 1895 <= self._year <= 1930:
+            # Anniversary of the capture of Rome.
+            self._add_holiday_sep_20(tr("Anniversario della Presa di Roma"))
+
+        # Established by Law 151 of Oct 8, 2025.
+        if self._year >= 2026:
+            self._add_holiday_oct_4(
+                # Saint Francis of Assisi, Patron Saint of Italy.
+                tr("Festa nazionale di San Francesco d'Assisi, patrono d'Italia")
+            )
+
+        # Established by Royal Decree-Law 1779 of Oct 21, 1926.
+        # Suspended by Royal Decree-Law 781 of Jul 24, 1941.
+        if 1926 <= self._year <= 1940:
+            # Anniversary of the March on Rome.
+            self._add_holiday_oct_28(tr("Anniversario della Marcia su Roma"))
+
+        # Established by Royal Decree 5342 of Oct 17, 1869.
 
         # All Saints' Day.
-        self._add_all_saints_day("Tutti i Santi")
-
-        if self._year <= 1976:
-            # National Unity and Armed Forces Day.
-            self._add_holiday_nov_4("Giornata dell'Unità Nazionale e delle Forze Armate")
-
-        # Immaculate Conception.
-        self._add_immaculate_conception_day("Immacolata Concezione")
+        self._add_all_saints_day(tr("Ognissanti"))
+
+        # Established by Royal Decree-Law 1354 of Oct 23, 1922.
+        # Suspended by Royal Decree-Law 781 of Jul 24, 1941.
+        # Restored by Legislative Decree 185 of Apr 22, 1946.
+        # Renamed by Law 260 of May 27, 1949.
+        # Made movable by Law 54 of Mar 5, 1977.
+        if 1922 <= self._year <= 1940 or self._year >= 1946:
+            name = (
+                # National Unity Day.
+                tr("Giorno dell'unità nazionale")
+                if self._year >= 1950
+                # Victory Day.
+                else tr("Anniversario della Vittoria")
```

Comment:
Shifted to November 6th for 1927 by Royal Decree-Law 1922 of Oct 23, 1927
https://www.gazzettaufficiale.it/eli/gu/1927/10/25/247/sg/pdf

Shifted to November 3rd for 1929 by Royal Decree-Law 1827 of Oct 13, 1929
https://www.gazzettaufficiale.it/eli/gu/1929/10/22/246/sg/pdf